src/ng/http.js

```diff
@@ -1293,7 +1297,7 @@ function $HttpProvider() {
           reqHeaders[(config.xsrfHeaderName || defaults.xsrfHeaderName)] = xsrfValue;
         }
 
-        $httpBackend(config.method, url, reqData, done, reqHeaders, config.timeout,
+        $httpBackend(config.method, (url === $sce.valueOf(origUrl) ? origUrl : url), reqData, done, reqHeaders, config.timeout,
```

Comment:
> The trouble is that many JSONP requests are going to be parameterised, right?

There are two types of "parameterized". If the user trusts the URL (including the parameters - e.g. `$http.jsonp($sce.trustAsResourceUrl('url?with=params', ...)`), then it could be fine.
What they can't do, is pass the parameters as `config.params` (e.g. `$http.jsonp('url', {params: {...}})`).

--
> So perhaps we should only allow string urls (i.e. you cannot explicitly trust them), which is a massive breaking change for JSONP requests.

This particular thing is not a breaking change, since we are already restricting to string URLs, right?